pkg/cmd/browse/browse.go

```diff
@@ -181,53 +181,46 @@ func addArg(opts *BrowseOptions, url string, branchName string) (exitCode, strin
 
 	if isNumber(opts.SelectorArg) {
 		url += "/issues/" + opts.SelectorArg
-		return exitSuccess, url
+		return exitNonUrlSuccess, url
 	}
 
 	arr := parseFileArg(opts)
 	if len(arr) > 1 {
-		return exitSuccess, url + "/tree/" + branchName + "/" + arr[0] + "#L" + arr[1]
+		return exitUrlSuccess, url + "/tree/" + branchName + "/" + arr[0] + "#L" + arr[1]
 	}
 
-	return exitSuccess, url + "/tree/" + branchName + "/" + arr[0]
+	return exitUrlSuccess, url + "/tree/" + branchName + "/" + arr[0]
 }
 
 func parseFileArg(opts *BrowseOptions) []string {
 	arr := strings.Split(opts.SelectorArg, ":")
 	return arr
 }
 
-func printExit(exit exitCode, cmd *cobra.Command, opts *BrowseOptions, url string) {
+func printExit(exit exitCode, cmd *cobra.Command, opts *BrowseOptions, url string) error {
```

Comment:
@vilmibm We are now returning an `error` from our `printExit` function, which then returns from `openInBrowser` to `RunE:`. 

We were wondering if we could keep the `exitCode` so we can print the appropriate response? This just makes it easier for us to manage, as all the output is in one spot. If this isn't a good strategy, do you think you could point us in the right direction to start restructuring? Thanks

Reply:
The code in `cmd/gh/main.go` will print out whatever error it sees:

https://github.com/cli/cli/blob/trunk/cmd/gh/main.go#L190

please do replace code like this:

```go
if !inRepo(err) {
  return printExit(exitNotInRepo, cmd, opts, "")
}
```

with code like this:

```go

if !inRepo(err) {
  return errors.New("change directory to a repository to open in browser\n%s")
}
```

better yet, you could go even further with that one and just do:

```go
baseRepo, err := opts.BaseRepo()
if err != nil {
  return fmt.Errorf("unable to determine base repository: %w", err)
}
```

you'll see that pattern all through a Go codebase. `fmt.Errorf` takes a format string and an error and wraps the error's message with further information.

how does that sound?

> Also, is there a way to select a section of code to comment like how you did? Thanks

You can click on the + sign at a line in the files view then drag down to select a range of lines to comment on.

Reply:
> You can click on the + sign at a line in the files view then drag down to select a range of lines to comment on.
Ah gotcha, thank you for that

Also I see what you mean now by looking at other print statements around the project. We'll be sure to change the output of our command to match the format that's already there.